File: rust/sdl2/src/game_objects/apple.rs
use sdl2::pixels::Color;
use sdl2::rect::{Rect};
use sdl2::render::{Canvas};
use sdl2::video::{Window};
use crate::game_objects::constants::{SPRITE_SIZE};

pub struct Apple {
    pub x: i32,
    pub y: i32
}

impl Apple {
    pub fn draw(&mut self, canvas: &mut Canvas<Window>) {
        canvas.set_draw_color(Color::RGB(255,0,0));
        let width = (SPRITE_SIZE - 1) as u32;
        let height = (SPRITE_SIZE - 1) as u32;
        canvas.fill_rect(
            Rect::new(self.x * SPRITE_SIZE + 1, self.y * SPRITE_SIZE + 1, width, height))
            .expect("Failed to draw the apple");
    }
}

File: golang/pixel/objects/apple.go
package objects

import (
	"image/color"
	"snake/constants"
	"snake/interfaces"
)

type Apple struct {
	Position Vector
	Width    int
	Height   int
	Color    color.RGBA
}

func (a *Apple) Move(x, y int) {
	a.Position.X = x
	a.Position.Y = y
}

func (a *Apple) Draw(renderer interfaces.Renderer) {
	x := a.Position.X * constants.SpriteSize
	y := a.Position.Y * constants.SpriteSize
	renderer.RenderRectangle(x, y, x+a.Width, y+a.Height, a.Color)
}
